src/cwrapper.h

```diff
@@ -0,0 +1,98 @@
+#ifndef CWRAPPER_H
+#define CWRAPPER_H
+
+#include <gmp.h>
+
+#ifdef __cplusplus
+#include "basic.h"
+#include "constants.h"
+namespace CSymPy {
+    class CWrapper {
+        public:
+            RCP<const Basic> ptr;
+            CWrapper() {
+                ptr = zero;
+            }
+    };
+}
+using CSymPy::CWrapper;
+
+extern "C" {
+
+#else
+typedef struct CWrapper CWrapper;
+#endif
```

Comment:
If it works, we should try to do it that way, I think it will be much cleaner and easier to understand for people that just want to call csympy from C.